### ISSUE TITLE
Pterodactyl fails to put default values when creating server and crashes Wings

### ISSUE THREAD

**Issue by Rems19** [pterodactyl/panel#2261]
**Background:**
* Panel or Daemon: Both
* Version of Panel/Daemon: v1.0.0-rc.1 (for both)
* Server's OS: Debian 10
* Your Computer's OS & Browser: Windows10, Edge (Chromium)

**Describe the bug**
When omitting certain parameters from the server creation page (or the server creation API call), the panel fails to put default because of the way `Arr:get($array, $key, $default = null)` works. If the field is present (which it is because it comes from a form) and is `null`, it returns `null` and not the default. When Pterodactyl sends the server to Wings, Wings crashes (panic) while trying to read the variables in environment/settings.go, at `return val.(string)` because `val` is `nil`.

Wings' crash logs:
```
DEBUG: [Aug 24 16:51:19.772] creating required server data directory path=/var/lib/pterodactyl/volumes/636a0f90-d1ba-4fff-b993-be3ffa40bf02 process=installer server=636a0f90-d1ba-4fff-b993-be3ffa40bf02
DEBUG: [Aug 24 16:51:19.772] creating required environment for server instance process=installer server=636a0f90-d1ba-4fff-b993-be3ffa40bf02
DEBUG: [Aug 24 16:51:20.974] pulling docker image... this could take a bit of time image=quay.io/pterodactyl/core:java
DEBUG: [Aug 24 16:51:20.975] Getting variable:
DEBUG: [Aug 24 16:51:20.975]   -                       key=P_SERVER_ALLOCATION_LIMIT
DEBUG: [Aug 24 16:51:20.975]   -                       value=<nil>
DEBUG: [Aug 24 16:51:20.975]   -                       valueType=<nil>
panic: interface conversion: interface {} is nil, not string

goroutine 89 [running]:
github.com/pterodactyl/wings/environment.Variables.Get(0xc0006c9410, 0xc000443980, 0x19, 0x19, 0x1)
        /mnt/c/Users/remil/Documents/Dev/wings/environment/settings.go:131 +0x5d4
github.com/pterodactyl/wings/environment.(*Configuration).EnvironmentVariables(0xc0004be8c0, 0xc0001babc8, 0xc0006c9fb0, 0x0)
        /mnt/c/Users/remil/Documents/Dev/wings/environment/config.go:80 +0x553
github.com/pterodactyl/wings/environment/docker.(*Environment).variables(0xc0001ba3f0, 0xc0006c9fb0, 0xc0005fe900, 0x3)
        /mnt/c/Users/remil/Documents/Dev/wings/environment/docker/container.go:208 +0x36
github.com/pterodactyl/wings/environment/docker.(*Environment).Create(0xc0001ba3f0, 0x0, 0xc0001fd030)
        /mnt/c/Users/remil/Documents/Dev/wings/environment/docker/container.go:151 +0x26e
github.com/pterodactyl/wings/installer.(*Installer).Execute(0xc000010268)
        /mnt/c/Users/remil/Documents/Dev/wings/installer/installer.go:121 +0x793
github.com/pterodactyl/wings/router.postCreateServer.func1(0xc000010268)
        /mnt/c/Users/remil/Documents/Dev/wings/router/router_system.go:60 +0x2f
created by github.com/pterodactyl/wings/router.postCreateServer
        /mnt/c/Users/remil/Documents/Dev/wings/router/router_system.go:59 +0x256
```
The line numbers in settings.go might differ a bit since I added debug on a clone to track down the issue.

**To Reproduce**
1. Go to '/admin/servers/new' on the panel
2. Fill in the form but remove the default 0 value from the "Allocation Limit" field
2. Click on 'Create Server'
4. Wings crashes and the server is shown by the panel as "installing". You can't start Wings again before forcibly deleting the server.

**Expected behavior**
In the panel code, in app/Services/SeverCreationService.php, in the `createModel(array $data): Server` function, the default values should be set if the value provided was `null`, and not only if the key is completely absent.
In addition, and that's why I said it concerned both the panel and Wings, the latter should not crash when wrong data is sent.

**Comment by DaneEveritt** [comment]
Thanks for the detailed error report!

**Comment by Rems19** [comment]
> Thanks for the detailed error report!

You're welcome! I might try to make a PR, at least for the panel side. I'm not confident enough to try to make one in Go yet.

**Comment by schrej** [comment]
Just go (lol) for it. The worst that can happen is that the wings PR gets closed and dismissed in favour of a better implementation that you can learn from.

**Comment by DaneEveritt** [comment]
There is some additional logic we probably need to do anyways, since it is technically possible for a variable to have no default value either. I suppose in that instance it can send an empty string along for now which should at least fix it. Wings will need slightly more work to handle the strings correctly, likely a custom json unmarshal function.

**Comment by schrej** [comment]
Looking at how Arr::get is implemented, isn't that a Laravel bug?
https://github.com/laravel/framework/blob/7.x/src/Illuminate/Support/Arr.php#L292

**Comment by Rems19** [comment]
I was wondering about it but that may as well be an implementation choice. If a key is present in a key,value array , it's not that surprising that it returns the value as is, even when it's null. But I agree that it's confusing, the PHPDoc should be clearer about that.

**Comment by schrej** [comment]
Well, the left part of L297 shouldn't be reachable. If the key exists it would return in the previous if statement already.

**Comment by Rems19** [comment]
Indeed, I misread the false check.

**Comment by matthewpi** [comment]
https://github.com/pterodactyl/panel/pull/2262

**Comment by Aeris1One** [comment]
Still having this bug, I just upgraded my 0.7 instance to 1.0 rc2 (in which this is fixed) and when adding the node I had this panic error

**Comment by DaneEveritt** [comment]
That is a different issue, please open a new issue and include those logs.
